test/e2e/step_definitions/utils.js

```diff
@@ -0,0 +1,15 @@
+const { promisify } = require('util')
+
+const sleep = promisify(setTimeout)
+
+module.exports.waitForCondition = async (evaluateCondition, timeout = 1000) => {
+  let remainingTime = timeout
+  while (!evaluateCondition()) {
+    if (remainingTime > 0) {
+      await sleep(50)
+      remainingTime -= 50
+    } else {
+      throw new Error('Condition not fulfilled until timeout')
```

Comment:
I think this would be clearer as 'Timed out and condition not fulfilled'

Reply:
I don't quite agree. The main point is that condition was not fulfilled, not that it timed out.

Reply:
The `until` seems odd to me.  Maybe:
`Condition not fulfillled, waited ${timeout}ms`

Reply:
Updated